{
  final ModelNode addOp=getAddDeploymentScannerOp();
  final ModelNode resultOfRollback=executeAndRollbackOperation(addOp);
  assertEquals("Unexpected result from invoking an operation which was supposed to fail","failed",resultOfRollback.get("outcome").asString());
  addDeploymentScanner();
  try {
    assertTestDeploymentScannerResourceExists();
    final ModelNode removeOp=getRemoveDeploymentScannerOp();
    final ModelNode resultOfRemoveRollback=executeAndRollbackOperation(removeOp);
    assertEquals("Unexpected result from invoking an operation which was supposed to fail","failed",resultOfRemoveRollback.get("outcome").asString());
    assertTestDeploymentScannerResourceExists();
  }
  finally {
    try {
      removeDeploymentScanner();
    }
 catch (    Throwable t) {
      logger.info("Removing test deployment scanner failed with exception",t);
    }
  }
}
